File: 2021/misc/02.pl
#!/usr/bin/env perl
use 5.010;
while (<>) {
    /forward (\d+)/ and $x += $1, $y2 += ($aim * $1);
    /down (\d+)/    and $y += $1, $aim += $1;
    /up (\d+)/      and $y -= $1, $aim -= $1;
}
say $x * $y;
say $x * $y2;

File: 2021/misc/02.rb
#!/usr/bin/env ruby
x = y = y2 = aim = 0
ARGF.each do |line|
  x += Integer($1) and y2 += aim * Integer($1) if line =~ /forward (\d+)/
  y += Integer($1) and aim += Integer($1)      if line =~ /down (\d+)/
  y -= Integer($1) and aim -= Integer($1)      if line =~ /up (\d+)/
end
puts x * y
puts x * y2
